Story/StoryOnlineMenu.cs

```diff
@@ -134,90 +116,168 @@ public override void ShutDownProcess()
             base.ShutDownProcess();
         }
 
-
-        private void SteamSetup()
+        private void UpdatePlayerList()
         {
+            playerButtons.Do(x => StoryMenuHelpers.RemoveMenuObjects(x));
+            playerButtons.Clear();
+
+            var pos = new Vector2(194, 553);
 
-            List<PlayerInfo> players = new List<PlayerInfo>();
-            foreach (OnlinePlayer player in OnlineManager.players)
+            foreach (var playerInfo in MatchmakingManager.instance.playerList)
             {
-                CSteamID playerId;
-                if (player.id is LocalMatchmakingManager.LocalPlayerId)
-                {
-                    playerId = default;
-                }
-                else
+                pos -= new Vector2(0, 38);
+                var btn = new SimplerButton(this, this.pages[0], playerInfo.name, pos, new(110, 30));
+                btn.OnClick += (_) =>
                 {
-                    playerId = (player.id as SteamMatchmakingManager.SteamPlayerId).steamID;
-                }
-                players.Add(new PlayerInfo(playerId, player.id.name));
+                    if (playerInfo.id != default)
+                    {
+                        SteamFriends.ActivateGameOverlayToWebPage($"https://steamcommunity.com/profiles/{playerInfo.id}");
+                    }
+                };
+                this.pages[0].subObjects.Add(btn);
             }
-            this.players = players.ToArray();
         }
 
-        private void UpdatePlayerList()
+        private void OnlineManager_OnPlayerListReceived(PlayerInfo[] players)
         {
-            for (int i = 0; i < playerButtons.Length; i++)
+            UpdatePlayerList();
+        }
+
+        public int GetCurrentlySelectedOfSeries(string series) => series switch
+        {
+            "scugButtons" => !RainMeadow.rainMeadowOptions.SlugcatCustomToggle.Value ? -1 : slugcatPageIndex,
+            _ => -1,
+        };
+
+        public void SetCurrentlySelectedOfSeries(string series, int to)
+        {
+            switch (series)
             {
-                var playerbtn = playerButtons[i];
-                playerbtn.RemoveSprites();
-                this.pages[0].RemoveSubObject(playerbtn);
+                case "scugButtons":
+                    slugcatPageIndex = to;
```

Comment:
I was expecting to see RemoveExcessStoryObjects called here again for when the page update triggers a new slugcat, but I only see it called when the client first joins the lobby. Can you verify?

Reply:
I changed RemoveExcessStoryObjects to remove for all pages

Reply:
Awesome